cmd/acra-keys/keys/list-keys_test.go

```diff
@@ -74,6 +83,165 @@ func TestPrintKeysJSON(t *testing.T) {
 	}
 }
 
+func TestListHistoricalKeysV1(t *testing.T) {
+	clientID := []byte("testclientid")
+	timesToGenerateHistoricalKeys := 3
+	keyloader.RegisterKeyEncryptorFabric(keyloader.KeystoreStrategyEnvMasterKey, env_loader.NewEnvKeyEncryptorFabric(keystore.AcraMasterKeyVarName))
+
+	masterKey, err := keystore.GenerateSymmetricKey()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	flagSet := flag.NewFlagSet(CmdMigrateKeys, flag.ContinueOnError)
+	keyloader.RegisterCLIParametersWithFlagSet(flagSet, "", "")
+
+	err = flagSet.Set("keystore_encryption_type", keyloader.KeystoreStrategyEnvMasterKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Setenv(keystore.AcraMasterKeyVarName, base64.StdEncoding.EncodeToString(masterKey))
+
+	dirName := t.TempDir()
+	if err := os.Chmod(dirName, 0700); err != nil {
+		t.Fatal(err)
+	}
+
+	listCMD := &ListKeySubcommand{
+		CommonKeyStoreParameters: CommonKeyStoreParameters{
+			keyDir: dirName,
+		},
+		CommonKeyListingParameters: CommonKeyListingParameters{
+			historicalKeys: true,
+		},
+		FlagSet: flagSet,
+	}
+
+	store, err := openKeyStoreV1(listCMD)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err = store.GenerateDataEncryptionKeys(clientID); err != nil {
+		t.Fatal(err)
+	}
+
+	for i := 0; i < timesToGenerateHistoricalKeys; i++ {
+		if err = store.GenerateDataEncryptionKeys(clientID); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	pubKeysEntries, err := os.ReadDir(filepath.Join(dirName, string(clientID)+"_storage.pub.old"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pubKeysTimes := make([]time.Time, 0, timesToGenerateHistoricalKeys)
+	for _, entry := range pubKeysEntries {
+		entryTime, err := time.Parse(filesystem.HistoricalFileNameTimeFormat, entry.Name())
+		if err != nil {
+			t.Fatal(err)
+		}
+		pubKeysTimes = append(pubKeysTimes, entryTime)
+	}
+
+	privateKeysEntries, err := os.ReadDir(filepath.Join(dirName, string(clientID)+"_storage.old"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	privateKeysTimes := make([]time.Time, 0, timesToGenerateHistoricalKeys)
+	for _, entry := range privateKeysEntries {
+		entryTime, err := time.Parse(filesystem.HistoricalFileNameTimeFormat, entry.Name())
+		if err != nil {
+			t.Fatal(err)
+		}
+		privateKeysTimes = append(privateKeysTimes, entryTime)
+	}
+
+	descriptions, err := store.ListHistoricalKeys()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(descriptions) != 2*timesToGenerateHistoricalKeys {
+		t.Fatal("Expect exact number of historical keys description")
+	}
+
+	for i := 0; i < timesToGenerateHistoricalKeys; i++ {
+		if descriptions[i].CreationTime.String() != privateKeysTimes[i].String() {
+			t.Fatalf("Not expected creation time of historical private key, %s not equal %s", descriptions[i].CreationTime.String(), privateKeysTimes[0].String())
+		}
+	}
+
+	for i := 0; i < timesToGenerateHistoricalKeys; i++ {
+		if descriptions[i+3].CreationTime.String() != pubKeysTimes[i].String() {
```

Comment:
```suggestion
		if descriptions[i+timesToGenerateHistoricalKeys].CreationTime.String() != pubKeysTimes[i].String() {
```

plus we can merge two cycles into one and add comment that public keys stored after private keys in the descriptions

Reply:
and can we compare content of the keys? at least public keys which are not encrypted to be sure that they returned properly according to names?